pkg/k8shandler/kibana/route.go

```diff
@@ -125,59 +130,59 @@ func (clusterRequest *KibanaRequest) createOrUpdateKibanaRoute() error {
 		}
 	}
 
+	return nil
+}
+
+func (clusterRequest *KibanaRequest) createOrUpdateKibanaConsoleLinks() error {
+	cluster := clusterRequest.cluster
+
 	kibanaURL, err := clusterRequest.GetRouteURL("kibana")
 	if err != nil {
 		return err
 	}
 
-	sharedConfig := createSharedConfig(cluster.Namespace, kibanaURL, kibanaURL)
-	utils.AddOwnerRefToObject(sharedConfig, getOwnerRef(cluster))
+	consoleAppLogsLink := NewConsoleLink(AppLogsConsoleLinkName, cluster.GetNamespace(), kibanaURL)
+	utils.AddOwnerRefToObject(consoleAppLogsLink, getOwnerRef(cluster))
 
-	err = clusterRequest.Create(sharedConfig)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return fmt.Errorf("Failure creating Kibana route shared config: %v", err)
+	if err := clusterRequest.createOrUpdateConsoleLink(consoleAppLogsLink); err != nil {
+		return fmt.Errorf("Failure creating or updating console app logs link for kibana CR %q: %v", cluster.Name, err)
 	}
 
-	sharedRole := NewRole(
-		"sharing-config-reader",
-		cluster.Namespace,
-		NewPolicyRules(
-			NewPolicyRule(
-				[]string{""},
-				[]string{"configmaps"},
-				[]string{"sharing-config"},
-				[]string{"get"},
-			),
-		),
-	)
-
-	utils.AddOwnerRefToObject(sharedRole, getOwnerRef(clusterRequest.cluster))
+	consoleInfraLogsLink := NewConsoleLink(InfraLogsConsoleLinkName, cluster.GetNamespace(), kibanaURL)
+	utils.AddOwnerRefToObject(consoleInfraLogsLink, getOwnerRef(cluster))
 
-	err = clusterRequest.Create(sharedRole)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return fmt.Errorf("Failure creating Kibana route shared config role for %q: %v", cluster.Name, err)
+	if err := clusterRequest.createOrUpdateConsoleLink(consoleInfraLogsLink); err != nil {
+		return fmt.Errorf("Failure creating or updating console infra logs link for kibana CR %q: %v", cluster.Name, err)
 	}
 
-	sharedRoleBinding := NewRoleBinding(
-		"openshift-logging-sharing-config-reader-binding",
-		cluster.Namespace,
-		"sharing-config-reader",
-		NewSubjects(
-			NewSubject(
-				"Group",
-				"system:authenticated",
-			),
-		),
-	)
-
-	utils.AddOwnerRefToObject(sharedRoleBinding, getOwnerRef(clusterRequest.cluster))
+	return nil
+}
 
-	err = clusterRequest.Create(sharedRoleBinding)
+func (clusterRequest *KibanaRequest) createOrUpdateConsoleLink(desired *consolev1.ConsoleLink) error {
```

Comment:
Consider writing an ApplyConsoleLink() function similar to Apply* functions in github.com/library-go/pkg/operator/resource/resourceapply that would just compare the routes' specs.

In the console we have our own func for that, check https://github.com/openshift/console-operator/blob/master/pkg/console/controllers/clidownloads/controller.go#L222

Reply:
We should consider providing this to `library-go` for our custom resources.

Reply:
Story in our backlog.
https://issues.redhat.com/browse/CONSOLE-2179